src/plugin.js

```diff
@@ -223,9 +190,45 @@ const perSourceBehaviors = function() {
     VERSION: '__VERSION__'
   };
 
-  // Add the per-source event binding methods to this player.
-  this.onPerSrc = onPerSrc;
-  this.onePerSrc = onePerSrc;
+  /**
+   * Bind an event listener on a per-source basis.
+   *
+   * @function onPerSrc
+   * @param  {String|Array|Component|Element} first
+   *         The event type(s) or target Component or Element.
+   *
+   * @param  {Function|String|Array} second
+   *         The event listener or event type(s) (when `first` is target).
+   *
+   * @param  {Function} third
+   *         The event listener (when `second` is event type(s)).
+   *
+   * @return {Player}
+   */
+  this.onPerSrc = createPerSrcBinder();
+
+  /**
+   * Bind an event listener on a per-source basis. This listener can only
+   * be called once.
+   *
+   * @function onePerSrc
+   * @param  {String|Array|Component|Element} first
+   *         The event type(s) or target Component or Element.
+   *
+   * @param  {Function|String|Array} second
+   *         The event listener or event type(s) (when `first` is target).
+   *
+   * @param  {Function} third
+   *         The event listener (when `second` is event type(s)).
+   *
+   * @return {Player}
+   */
+  this.onePerSrc = createPerSrcBinder(true);
+
+  // Clear out perSrcListeners cache on player dispose.
+  this.on('dispose', () => {
+    perSrcListeners.length = 0;
```

Comment:
I wonder if it would be better to have this event listener listen for the sourcechanged and dispose events. Then we can do the same type of empty on preSrcListeners for both.